parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveType.java

```diff
@@ -155,6 +181,14 @@ public void addValueToPrimitiveConverter(
       public <T, E extends Exception> T convert(PrimitiveTypeNameConverter<T, E> converter) throws E {
         return converter.convertBINARY(this);
       }
+
+      @Override
+      PrimitiveComparator<?> comparator(OriginalType logicalType) {
+        if (logicalType == OriginalType.DECIMAL) {
+          return PrimitiveComparator.SIGNED_BINARY_COMPARATOR;
+        }
+        return PrimitiveComparator.LEXICOGRAPHICAL_BINARY_COMPARATOR;
```

Comment:
I would prefer to be explicit about the logical types where LEXICOGRAPHICAL_BINARY_COMPARATOR is returned, so that if new logical types get added in the future that require different handling they won't automatically fall into the default case.